### PR TITLE
Refactor ShareFromCache to use variadic generics

### PR DESCRIPTION
**Pull request by aim2120**
## Description

Removes the manual n-Hashable APIs for `shareFromCache` in favor of variadic generics.

Includes some test additions and improvements for further validation.

### CODE REVIEW

File: Sources/Afluent/Workers/ShareFromCache.swift [hunk 1]
```diff
@@ -62,213 +62,11 @@ extension AsynchronousUnitOfWork {
...
-        hasher.combine(k0)
-        hasher.combine(k1)
-        hasher.combine(k2)
-        hasher.combine(k3)
-        hasher.combine(k4)
-        hasher.combine(k5)
-        hasher.combine(k6)
-        hasher.combine(k7)
-        hasher.combine(k8)
-        hasher.combine(🐶)
```
Comment (aim2120) [inline Sources/Afluent/Workers/ShareFromCache.swift:271]: 🫡 you served your purpose well

**Comment by Tyler-Keith-Thompson** [comment]
I love this refactor! However, I think we now find ourselves in an interesting position. It appears that the library currently supports Swift 5.8+ and I don't believe this would compile with 5.8. 

So the question is, do we want to drop that support. I believe that Swift versions and generally better adopted than OS versions across the industry, so on the surface supporting back to 5.8 is an easy choice. This probably ties into the other PR and the need to have more robust testing for everything Afluent supports

**Comment by Tyler-Keith-Thompson** [comment]
Update: Turns out 5.8 was broken for a bit already. I'm going to remove the Package.swift definition for it and remove it from the new matrix I added

File: Sources/Afluent/Workers/ShareFromCache.swift (hunk 1 above)
Comment (Tyler-Keith-Thompson) [reply Sources/Afluent/Workers/ShareFromCache.swift:271]: Biggest downside of this refactor by far

**Review by Tyler-Keith-Thompson — APPROVED** [review]

**Pull request merged**